### DISCUSSION TITLE
Config options for opacity/blur for unfocused windows

### DISCUSSION THREAD

**Discussion by lilyball** [ghostty-org/ghostty#2650] [Feature Requests, Ideas]
Ghostty has `background-opacity` and `background-blur-radius` options, but these apply to all windows. It would be great to have `inactive-background-opacity` and `inactive-background-blur-radius` (or e.g. `background-opacity = inactive=value`, or `focused-background-opacity`, or `background-opacity = focused=value`) that would apply to any window that is not focused, and if the app itself loses focus then that means it applies to all windows.

This is to mirror an Apple Terminal.app feature that does the same thing. In Terminal.app profiles, when you select the color well to configure the background color, it has the opacity/blur sliders there, and a checkbox to enable separate settings for inactive windows. And when Terminal.app itself loses focus, all windows are inactive. Note that, to use AppKit terminology, this is the "main" window, not the "key" window; opening the preferences window (which is an `NSPanel`) does not make the active terminal window adopt the inactive styling. Ghostty does not currently have a preferences window, though perhaps the Quick Terminal window, being a floating window, should do the same thing and avoid causing the active terminal to adopt the inactive styling.

I've been using this setting in Apple Terminal.app for well over a decade and I find it invaluable to help tell at a glance which terminal window is active (or whether Terminal itself is active).

Here's a screenshot of Apple Terminal.app's preferences with the background color well selected:

<img width="665" alt="Screenshot of Apple Terminal preferences" src="https://github.com/user-attachments/assets/c4cabe23-4b78-4b54-8cee-630c962f991f">

**Comment by xim** [comment]
I also want something like this for unfocused windows, but ideally not blur nor opacity ; If you use blur, opacity or dimming, the text is likely to become less readable. What I want is to desaturate the the colors when the window isn't focused.

![image](https://github.com/user-attachments/assets/be8fc9ed-4c95-4d25-817c-92d3035b6cb7)

In this image, I have the same content in two terminals where one of them have a desaturation applied per the formula `sqrt(0.299*R^2 + 0.587*G^2 + 0.114*B^2)` to keep perceived luminance (and thus readability) as per option 2 of [some random SO answer](https://stackoverflow.com/questions/596216/formula-to-determine-perceived-brightness-of-rgb-color)...

**Comment by jadnhm** [comment]
I came here looking for a way to achieve this, maybe my favourite feature of `Terminal.app` and one of the main reasons I never switched away from it.
